app/api/update_email/route.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  generateEmailToken,
+  sendVerificationEmail,
+  storeTokenInDb,
+} from "@/utils/emailToken";
+import { NextRequest, NextResponse } from "next/server";
+
+export async function POST(req: NextRequest, res: NextResponse) {
+  const { userId, newEmail } = await req.json();
```

Comment:
This could let someone change someone elses email. You would need to grab the user/userId from the current users session. Also for mutating requests like this we use trpc. You could add this here: https://github.com/codu-code/codu/blob/develop/server/api/router/profile.ts

And you'll see the pattern we have for cleaning data with schemas and getting the user from `ctx.session.user`

Reply:
Yes followed this way now.